File: src/main/java/kg/musabaev/cluserizator/MainView.java
package kg.musabaev.cluserizator;

import de.saxsys.mvvmfx.InjectViewModel;
import de.saxsys.mvvmfx.JavaView;
import javafx.fxml.Initializable;
import javafx.scene.control.TreeItem;
import javafx.scene.control.TreeView;
import javafx.scene.layout.BorderPane;

import javax.inject.Inject;
import java.net.URL;
import java.util.ResourceBundle;
import java.util.concurrent.atomic.AtomicReference;

public class MainView extends BorderPane implements JavaView<MainViewModel>, Initializable {

    private final TreeItem<SeoKeywordModel> root = new TreeItem<>();
    private final TreeView<SeoKeywordModel> treeView = new TreeView<>(root);

    @InjectViewModel
    private MainViewModel mainViewModel;
    @Inject
    private TestService testService;

    @Inject
    public MainView(MainViewModel mainViewModel, TestService testService) {
        this.testService = testService;
    }

    public MainView() {
    }

    @Override
    public void initialize(URL url, ResourceBundle resourceBundle) {
        treeView.setEditable(true);
        treeView.setCellFactory(p -> new TextFieldTreeCellImpl());
        treeView.setShowRoot(false);


        super.setCenter(treeView);
        testService.consoleLog();

        FileHandler testCsvFileHandler = new TestCsvFileHandler();

        AtomicReference<Integer> i = new AtomicReference<>(1);
        testCsvFileHandler.getLinesCsv().forEach(line -> {
            String[] values = line.split(",");

            String[] otherMetas = new String[values.length - 1];
            System.arraycopy(values, 1, otherMetas, 0, values.length - 1);
            Integer id = i.get();
            String keyword = values[0];

            SeoKeywordModel seoKeyword = new SeoKeywordModel(id, keyword, otherMetas);
            i.getAndSet(i.get() + 1);

            root.getChildren().add(new TreeItem<>(seoKeyword));
            mainViewModel.getSeoKeywords().add(seoKeyword);
        });
//        initStyles();
    }

//    public void initStyles() {
//        tableView.getColumns().get(1)
//                .prefWidthProperty().bind(Bindings.subtract(
//                        getScene().getWindow().getWidth(),
//                        tableView.getColumns().get(0).widthProperty()));
//    }
}

File: src/main/kotlin/kg/musabaev/cluserizator/MainView.kt
package kg.musabaev.cluserizator

import de.saxsys.mvvmfx.InjectViewModel
import de.saxsys.mvvmfx.JavaView
import javafx.fxml.Initializable
import javafx.scene.control.TreeItem
import javafx.scene.control.TreeView
import javafx.scene.layout.BorderPane
import javafx.util.Callback
import java.net.URL
import java.util.*
import java.util.concurrent.atomic.AtomicReference
import javax.inject.Inject

class MainView : BorderPane, JavaView<MainViewModel>, Initializable {
    private val root = TreeItem<SeoKeywordModel>()
    private val treeView = TreeView(root)

    @InjectViewModel
    private lateinit var mainViewModel: MainViewModel

    @Inject
    private lateinit var testService: TestService

    @Inject
    constructor(mainViewModel: MainViewModel, testService: TestService) {
        this.mainViewModel = mainViewModel
        this.testService = testService
    }

    constructor()

    override fun initialize(url: URL?, resourceBundle: ResourceBundle?) {
        treeView.isEditable = true
        treeView.cellFactory = Callback { TextFieldTreeCellImpl() }
        treeView.isShowRoot = false


        super.setCenter(treeView)
        testService.consoleLog()

        val testCsvFileHandler: FileHandler = TestCsvFileHandler()

        val i = AtomicReference(1)
        testCsvFileHandler.getLinesCsv().forEach { line: String ->
            val values = line.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            val otherMetas = values.copyOfRange(1, values.lastIndex - 1)
            val id = i.get()
            val keyword = values[0]

            val seoKeyword = SeoKeywordModel(id, keyword, otherMetas)
            i.getAndSet(i.get() + 1)

            root.children.add(TreeItem(seoKeyword))
            mainViewModel.seoKeywords.add(seoKeyword)
        }
    }
}